fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/LinkHeaderProvider.java

```diff
@@ -49,18 +49,18 @@
  * @author whikloj
  * @since 2015-10-30
  */
-@Component
+@Provider
```

Comment:
`@Component` is for Spring creation/scanning. `@Provider` is for jax-rs resources. Was this change intentional? ...it appears to have broken the auto-injection functionality.

Reply:
I don't think it was intentionally - will fix.

Reply:
Yes - it was intentional.  This provider uses the jax-rs @Context parameter for injecting resources.   I changed it to @Provider to fix the problem which was that the dependencies weren't being injected.

Reply:
I just double-checked to make sure...
As far as I can tell, this change removes `LinkHeaderProvider` from any execution flow.
The previous logic was that `ContentExposingResource` would invoke the `HttpHeaderInjector`:
https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java#L691

..which looks for Spring-managed classes that implement `UriAwareHttpHeaderFactory`:
https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/api/HttpHeaderInjector.java#L81

By changing `LinkHeaderProvider`s annotation from `@Component` (Spring) to `@Provider` (jax-rs), the class is no longer found by `HttpHeaderInjector`... and therefore never invoked.

Reply:
As a side note, this class only uses the `@Context` annotation for injecting resources because you made that change in this PR :)

Reply:
There is a more fundamental issue, which may have been the reason for your introduction of this change, which how the `TransactionProvider` is now trying to be used. It is trying to be managed by both jax-rs and Spring:
https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/session/TransactionProvider.java#L38-L40

..which I do not think is possible. 

Additionally, the `TransactionProvider` implements `org.glassfish.hk2.api.Factory<Transaction>` in order to facilitate @Injecting Transactions into classes via HK2 (https://eclipse-ee4j.github.io/jersey.github.io/documentation/latest/ioc.html#d0e17033). It appears that an attempt was made to change this approach in https://github.com/fcrepo4/fcrepo4/pull/1559/files . Instead of Injecting `TransactionProvider`, I believe we should be Injecting `Transaction`... and allowing the TransactionProvider Factory to "provide" the Transaction.

I will look into...

Reply:
Issues should now be resolved with: https://github.com/dbernstein/fcrepo4/pull/8

Reply:
Merged.